tools/fuzzer/ast.cc

```diff
@@ -168,8 +168,17 @@ BinaryExpr::BinaryExpr(Expr lhs, BinOp op, Expr rhs)
     : lhs_(std::make_shared<Expr>(std::move(lhs))),
       rhs_(std::make_shared<Expr>(std::move(rhs))),
       op_(op) {}
+BinaryExpr::BinaryExpr(Expr lhs, BinOp op, Expr rhs, Type type_for_unit_testing)
```

Comment:
Let's call it just `result_type` or `expr_type`. It's not just for unit testing, it makes total sense to have it in the AST.